src/components/Firebase/Firebase.tsx

```diff
@@ -24,10 +24,18 @@ type FirebaseProviderProps = {
 };
 
 export function FirebaseProvider(props: FirebaseProviderProps) {
+  const [isInitialized, setIsInitialized] = useState(false);
+
   useEffect(() => {
     // Initialize Firebase
     firebase.initializeApp(props.config);
-  });
+    setIsInitialized(true);
+  }, [props.config]);
+
+  // TODO: Create spinner/loading bar while initializing datastore
```

Comment:
We should actually pass in a component for the loading state for easy modification and SoC